File: python/code_examples/08-async/main.js
function sleep(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

async function fetchValue() {
  await sleep(50);
  return 42;
}

async function main() {
  const a = await fetchValue();
  const results = await Promise.all([fetchValue(), fetchValue()]);
  console.log(a, results);
}

main();

File: python/code_examples/08-async/main.py
import asyncio


async def fetch_value() -> int:
    await asyncio.sleep(0.05)
    return 42


async def main() -> None:
    a = await fetch_value()
    results = await asyncio.gather(fetch_value(), fetch_value())
    print(a, results)


if __name__ == "__main__":
    asyncio.run(main())
